frontend/src/metabase-lib/lib/Dimension.ts

```diff
@@ -982,8 +986,10 @@ export class FieldDimension extends Dimension {
       { ...this._options, ...options },
       this._metadata,
       this._query,
-      this._fieldInstance && {
+      {
         _fieldInstance: this._fieldInstance,
+        _subDisplayName: this._subDisplayName,
+        _subTriggerDisplayName: this._subTriggerDisplayName,
```

Comment:
These two properties are added here https://github.com/metabase/metabase/blob/dac22382883b2da98fdddcf44abe7c07581e5401/frontend/src/metabase-lib/lib/Dimension.ts#L991 so when reinstantiating FieldDimensions with more options via the `withOptions` method, we need to retain them explicitly.